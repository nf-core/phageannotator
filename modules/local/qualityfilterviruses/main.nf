process QUALITYFILTERVIRUSES {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' :
        'biocontainers/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' }"

    input:
    tuple val(meta), path(viruses), path(proviruses), path(quality_summary)

    output:
    tuple val(meta), path("*.filtered.fna.gz")  , emit: filtered_viruses
    path "versions.yml"                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    quality_filter_viruses.py \\
        --viruses $viruses \\
        --proviruses $proviruses \\
        --quality_summary $quality_summary \\
        --output ${prefix}.filtered.fna \\
        $args

    gzip ${prefix}.filtered.fna

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(echo \$(biopython_version.py 2>&1))
        pandas: \$(echo \$(pandas_version.py 2>&1))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.filtered.fna
    gzip ${prefix}.filtered.fna

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(echo \$(biopython_version.py 2>&1))
        pandas: \$(echo \$(pandas_version.py 2>&1))
    END_VERSIONS
    """
}
