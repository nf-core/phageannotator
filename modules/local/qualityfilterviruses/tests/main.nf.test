nextflow_process {

    name "Test process: QUALITYFILTERVIRUSES"
    script "../main.nf"
    process "QUALITYFILTERVIRUSES"


    test("virus_fasta.gz & provirus_fasta.gz & quality_summary.tsv") {

        when {

            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.pipelines_testdata_base_path + 'modules/local/quality_filter_viruses/viruses.fna.gz', checkIfExists: true),
                    file(params.pipelines_testdata_base_path + 'modules/local/quality_filter_viruses/proviruses.fna.gz', checkIfExists: true),
                    file(params.pipelines_testdata_base_path + 'modules/local/quality_filter_viruses/quality_summary.tsv', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("virus_fasta.gz & provirus_fasta.gz & quality_summary.tsv - stub") {

        options "-stub"

        when {

            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.pipelines_testdata_base_path + 'modules/local/quality_filter_viruses/viruses.fna.gz', checkIfExists: true),
                    file(params.pipelines_testdata_base_path + 'modules/local/quality_filter_viruses/proviruses.fna.gz', checkIfExists: true),
                    file(params.pipelines_testdata_base_path + 'modules/local/quality_filter_viruses/quality_summary.tsv', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
