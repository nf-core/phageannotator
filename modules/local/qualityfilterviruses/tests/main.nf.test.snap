{
    "virus_fasta.gz & provirus_fasta.gz & quality_summary.tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fna.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4065a8af6c710ebcd076fbcbabc91f17"
                ],
                "filtered_viruses": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fna.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4065a8af6c710ebcd076fbcbabc91f17"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-19T17:04:08.487686508"
    },
    "virus_fasta.gz & provirus_fasta.gz & quality_summary.tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fna.gz:md5,e807764af50f14c5ca5828c51869c113"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4065a8af6c710ebcd076fbcbabc91f17"
                ],
                "filtered_viruses": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fna.gz:md5,e807764af50f14c5ca5828c51869c113"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4065a8af6c710ebcd076fbcbabc91f17"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-19T17:03:57.878250244"
    }
}
